src/ActionManager.php

```diff
@@ -23,6 +32,22 @@ public function __construct(array $designPatterns = [])
         $this->setDesignPatterns($designPatterns);
     }
 
+    /**
+     * @param class-string<JobDecorator> $jobDecoratorClass
+     */
+    public static function useJobDecorator(string $jobDecoratorClass): void
+    {
+        static::$jobDecorator = $jobDecoratorClass;
+    }
+
+    /**
+     * @param class-string<JobDecorator&ShouldBeUnique> $uniqueJobDecoratorClass
```

Comment:
These are the type constraints that Laravel Actions actually expects this class to have. I double checked this with a local repository to see that PHPStan would actually be able to check for this correctly.